migrator/utils.lua

```diff
@@ -52,9 +52,30 @@ local function register_sharding_key(space_name, key)
     box.space._ddl_sharding_key:replace{space_name, key}
 end
 
+
+--- Checks are cartridge_roles enableb on server, returns true or false
+--- @param roles_list table array of role names
+---
+local function check_roles_enabled(roles_list)
+    local cur_roles = require('cartridge.lua-api.get-topology').get_enabled_roles_without_deps()
```

Comment:
Or the name `get_enabled_roles_without_deps` is actually confusing and in checks dependencies as well? (Since it looks like it works for dependency roles as well.)

Reply:
It returns all roles that showed in ui. I think that's enough. I coundn't find best api in cartridge for my goal.  cartridge('roles') functions work very strange.

<img width="995" alt="Снимок экрана 2024-01-25 в 20 00 55" src="https://github.com/tarantool/migrations/assets/26044679/00e39a15-9b4c-48f2-a3d1-b0fb3344460a">

```lua
admin:secret-cluster-cookie@localhost:3301> require('cartridge.lua-api.get-topology').get_enabled_roles_without_deps()
---
- - space-explorer
  - vshard-storage
  - crud-storage
...
```

May be there is way to do it withoud cartridge funcs

Reply:
yeah i suppose there is another way

```lua
confapplier.get_readonly('topology').replicasets[box.info.cluster.uuid].roles
```